### ISSUE TITLE
Siege objectives captured with vehicle do not reward points to the pilot

### ISSUE THREAD

**Issue by deathsythe47** [TheSil/base_enhanced#64]
You get zero points for capturing the first objective on Hoth, because the obj is being triggered by the ATST and not the player directly. The points should be awarded to the pilot of the triggering vehicle.

Btw, you DO get points for capturing the second obj(kill the generator) in ATST. That obj is detected fine.

**Comment by deathsythe47** [comment]
partial fix in 81bbef222c10010d67ca2fab5fd5656c82852ae7

still won't give points properly if a vehicle is required for the final obj of a map because of the way JA handles objs differently if they are either "the final obj" or "not the final obj" -- needs some more coding work to get this too

```
    if (final == 1 || goals_completed >= goals_required)
    {
        SiegeRoundComplete(team, client);
    }
    else
    {
        BroadcastObjectiveCompletion(team, objective, final, client);
    }
```

 luckily there are no siege maps (that I know of) with such an objective, so we're good for now

**Comment by deathsythe47** [comment]
Ok here is a proper fix that includes all objectives, both final and non-final.

e69b86967581d96a8407bafd096659886e478238 and i fixed a minor derp in that code in this one: c3f490bea6437eba3033d8e536f56aee210e2c67

**Comment by TheSil** [comment]
Thanks, i have slightly updated it. It should be tested though.
